src/record/record.js

```diff
@@ -628,14 +633,18 @@ Record.prototype._onTimeout = function( timeoutType ) {
  * @returns {void}
  */
  Record.prototype._destroy = function() {
- 	this._clearTimeouts();
- 	this._eventEmitter.off();
- 	this._resubscribeNotifier.destroy();
- 	this.isDestroyed = true;
- 	this.isReady = false;
- 	this._client = null;
-	this._eventEmitter = null;
-	this._connection = null;
+	if (this.usages > 0) {
+		this._init()
```

Comment:
This is the magic. We allow a record to be revive AFTER it's life cycle has finished.